source/infrastructure/pipeline/lib/pipeline-stack.ts

```diff
@@ -1,18 +1,22 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import CodeStarSource from "./code-star-source";
-import { Construct } from "constructs";
-import { CodeBuildStep, CodePipeline } from "aws-cdk-lib/pipelines";
-import { ComputeType, LinuxBuildImage } from "aws-cdk-lib/aws-codebuild";
-import { Stack, Stage } from "aws-cdk-lib";
-import { StringParameter } from "aws-cdk-lib/aws-ssm";
 import * as codebuild from "aws-cdk-lib/aws-codebuild";
-import { AwsInstanceSchedulerStack } from "../lib/aws-instance-scheduler-stack";
+import * as pipeslines from "aws-cdk-lib/pipelines"
+import CodeStarSource from "./code-star-source";
+import {Construct} from "constructs";
+import {Stack, Stage} from "aws-cdk-lib";
+import {StringParameter} from "aws-cdk-lib/aws-ssm";
+import {AwsInstanceSchedulerStack} from "../../instance-scheduler/lib/aws-instance-scheduler-stack";
 import {NagSuppressions} from "cdk-nag";
+import {E2eTestStack} from "./e2e-test-stack";
+import {Effect, PolicyStatement} from "aws-cdk-lib/aws-iam";
+import * as hubStackUtils from "../e2e-tests/utils/hub-stack-utils";
 
-const DEPLOY_STAGE_NAME = "test-deploy";
+const DEPLOY_STAGE_NAME = "Deployment-Test";
+const END_TO_END_STAGE_NAME = "End-to-End-Tests"
 const STACK_NAME = "InstanceScheduler";
+const TEST_RESOURCES_STACK_NAME = "InstanceSchedulerE2ETestResources";
```

Comment:
Any reason for these variables?

Reply:
only to avoid hardcoding strings like this in the middle of a bunch of other code

Reply:
But if it's not reused for anything, what's wrong with it being hard-coded?

Reply:
nothing, this is still hard-coded ;).

Defining it this way is just an attempt to treat it as a 'property' of the testing pipeline rather than an implementation detail buried in the weeds of everything else.

Reply:
I'm not saying that's a bad idea, but be careful doing things like that prematurely because I think that's the kind of thing that leads to proliferation of module-level variables you can see all over the application code. I think I would be a little more comfortable if they were instance variables of the pipeline stack class. I don't think them not being exported is as restrictive because I can always go in and `export` that variable, whereas I actually have to have pipeline stack instance to get at its instance variable.